src/template/page.zig

```diff
@@ -339,173 +409,141 @@ pub fn Page(comptime template: Template, comptime PageDataType: type) type {
+                    try out.writeAll(d);
+                } else if (directive.otherwise == .default) {
+                    try out.writeAll(directive.otherwise.default);
                 },
-                .Struct => {
-                    if (item) |itm| {
-                        try formatDirective(T, itm, ofs, html, out);
+                ?usize => {
+                    if (data) |us| {
+                        return try directive.formatTyped(usize, us, out);
                     }
                 },
-                else => unreachable,
+                else => {
+                    return try directive.formatTyped(T, data, out);
+                },
             }
         }
 
-        fn offsetArray(T: type, data: T, name: []const u8, ofs: []const Offset, html: []const u8, out: anytype) !void {
-            inline for (std.meta.fields(T)) |field| {
-                if (eql(u8, name, field.name)) {
-                    //std.debug.print("array found {s}\n", .{name});
-                    return switch (field.type) {
-                        []const u8 => try out.writeAll(@field(data, field.name)),
-                        []const []const u8 => try offsetArrayItem([]const u8, @field(data, field.name), ofs, html, out),
-                        else => {
-                            //std.debug.print("array found {}\n", .{@typeInfo(field.type)});
-                            switch (@typeInfo(field.type)) {
-                                .Pointer => |ptr| {
-                                    std.debug.assert(ptr.size == .Slice);
-                                    try offsetArrayItem(ptr.child, @field(data, field.name), ofs, html, out);
-                                },
-                                .Optional => |opt| {
-                                    if (opt.child == []const u8) {
-                                        return;
-                                        //return offsetDirective(field.type, @field(data, field.name), ofs[0].kind.directive, out);
-                                    }
-
-                                    try offsetOptionalItem(opt.child, @field(data, field.name), ofs, html, out);
-                                },
-                                else => {
-                                    std.debug.print("unexpected type {any}\n", .{field.type});
-                                    //unreachable;
-                                },
-                            }
-                        },
-                    };
-                }
-            } else {
-                std.debug.print("error generating page, field {s} is missing\n", .{name});
+        fn offsetOptionalItem(T: type, item: ?T, comptime ofs: []const Offset, html: []const u8, out: anytype) !void {
+            if (comptime T == ?[]const u8) return offsetDirective(T, item.?, ofs[0], out);
+            switch (@typeInfo(T)) {
+                .Int => std.debug.print("skipped int\n", .{}),
```

Comment:
What about float, enum, and union?

Reply:
both should be impossible, I'll make the following `else => unreachable` into `else => comptime unreachable` to prove that there's no way for any to reach this point in the code.